docs/sql/statements/create-repository.rst

```diff
@@ -1,303 +1,442 @@
+(HDFS).
+
+
+.. _sql-create-repo-hdfs-params:
+
+Parameters
+''''''''''
 
-.. rubric:: Parameters
+.. _sql-create-repo-hdfs-uri:
 
 **uri**
   | *Type:*    ``text``
-  | *Default:* default filesystem URI for the given Hadoop HDFS configuration
+  | *Default:* The default URI for the given HDFS configuration
 
-  HDFS uri of the form ``hdfs:// <host>:<port>/``.
+  HDFS URIs take the form of::
+
+      hdfs://<host>:<port>/
+
+.. _sql-create-repo-hdfs-security-principal:
 
 **security.principal**
   | *Type:*    ``text``
 
-  A qualified kerberos principal used to authenticate against HDFS.
+  A qualified Kerberos principal used to authenticate against HDFS.
+
+.. _sql-create-repo-hdfs-path:
 
 **path**
   | *Type:*    ``text``
 
-  HDFS filesystem path to where the data gets stored.
+  The HDFS file system path to use for snapshots.
+
+.. _sql-create-repo-hdfs-load_defaults:
 
 **load_defaults**
   | *Type:*    ``boolean``
   | *Default:* ``true``
 
-  Whether to load the default Hadoop Configuration.
+  Whether to load the default Hadoop configuration.
+
+.. _sql-create-repo-hdfs-conf-key:
 
 **conf.<key>**
   | *Type:*    various
 
-  Dynamic config values added to the Hadoop configuration.
+  Dynamic configuration values to be added to the Hadoop configuration.
+
+.. _sql-create-repo-hdfs-compress:
 
 **compress**
   | *Type:*    ``boolean``
   | *Default:* ``true``
 
-  Whether the metadata part of the snapshot should be compressed or not.
+  Whether CrateDB should compress the metadata part of the snapshot or not.
+
+  CrateDB does not compress the actual table data.
 
-  The actual table data is not compressed.
+.. _sql-create-repo-hdfs-chunk_size:
 
 **chunk_size**
   | *Type:*    ``bigint`` or ``text``
   | *Default:* ``null``
 
-  Defines the maximum size of a single file that gets created during snapshot
-  creation. If set to ``null`` big files will not be split into smaller chunks.
-  The chunk size can be either specified in bytes or using size value notation
-  (e.g. ``1g``, ``5m``, or ``9k``).
+  Defines the maximum size of any single file that comprises the snapshot. If
+  set to ``null``, CrateDB will not split big files into smaller chunks.  You
+  can specify the chunk size with units (e.g., ``1g``, ``5m``, or ``9k``). If
+  no unit is specified, the unit defaults to bytes.
+
 
-.. _ref-create-repository-types-s3:
+.. _sql-create-repo-s3:
 
 ``s3``
 ------
 
-A repository that stores its snapshot on the Amazon S3 service. When used in
-conjunction with Amazon IAM Roles, ``access_key`` and ``secret_key`` must be
-undefined and these credentials will then be loaded from the Amazon Container
-(e.g. EC2).
 
+An ``s3`` repository stores snapshot on the `Amazon Simple Storage Service`_
+(Amazon S3).
+
+.. NOTE::
+
+    If you are using Amazon S3 in conjunction with `IAM roles`_, the
+    ``access_key`` and ``secret_key`` parameters must be left undefined. The
+    IAM role will provide the necessary credentials when required.
```

Comment:
I think it is important to mention that when Amazon I AM Roles are used for authentication, that these roles must be provided by the underlying container where CrateDB is deployed.